src/plone/restapi/serializer/site.py

```diff
@@ -149,3 +150,34 @@ def serialize_blocks(self):
 
             blocks[id] = block_value
         return blocks
+
```

Comment:
I see at least one other difference between what serialize_blocks does here and what the "normal" blocks serializer does in https://github.com/plone/plone.restapi/blob/master/src/plone/restapi/serializer/blocks.py#L21 -- the latter also sets h.blockid on the handler.

Maybe it would be worth refactoring BlocksJSONFieldSerializer to use a function that could also be used from here. Then there would be less code duplication and less risk of it diverging over time.

Reply:
I tried to unify both in one place and write, say, a common base class for both. But one is a `IFieldSerializer` and the other one a `ISerializeToJson` so when trying to inherit one from the other, Zope complains of not having a proper zcml declaration.

Moreover the methods here are using `self.context` and `self.request` to get some subscribers to do transforms, so converting them to plain functions would mean to start passing context and request as parameters to those functions, what at least seems a bit ugly.

I am not favouring the "copy-paste" solution, just I don't see some other way to do it.

Reply:
Another option is to factor the relevant part out into a general function and call it from both classes/its methods.